scripts/index.js

```diff
@@ -0,0 +1,96 @@
+const titleInput = document.querySelector('#title_input');
+const authorInput = document.querySelector('#author_input');
+const bookList = document.querySelector('.book_list');
+const addBookButton = document.querySelector('#add_book');
+
+class Book {
+  constructor(id, title, authorName) {
+    this.id = id;
+    this.title = title;
+    this.authorName = authorName;
+  }
+}
+
+function init() {
+  if (!localStorage.getItem('books')) {
+    localStorage.setItem('books', JSON.stringify([]));
+  }
+}
+
+function getData(key) {
+  try {
+    return JSON.parse(localStorage.getItem(key));
+  } catch (error) {
+    return localStorage.getItem(key);
+  }
+}
+
+function templateBook(obj) {
+  const div = document.createElement('div');
+  div.classList.add('book');
+
+  const insideBook = `<p id="title">${obj.title}</p>
+            <p id="author">${obj.authorName}</p>
+            <button id="delete_book" type="submit" onClick="deleteBook(${obj.id})" data-id='${obj.id}'>delete</button>`;
+
+  div.innerHTML = insideBook;
+
+  bookList.append(div);
+}
+
+function displayBooks() {
+  const books = getData('books');
+
+  // clean the display books then before.
+  bookList.innerHTML = '';
+
+  if (books) {
+    books.forEach((book) => {
+      templateBook(book);
+    });
+  }
+}
+
+function updateStorage(book) {
+  const books = getData('books');
+  books.push(book);
```

Comment:
[OPTIONAL] Consider using methods that don't mutate the array.

<img width="407" alt="Screen Shot 2022-06-27 at 17 12 06" src="https://user-images.githubusercontent.com/82426682/176044355-2f603735-57c8-4f7a-8bb7-da4cb5ebd6f6.png">

Reply:
Thank you, yes this is better to use, i will use this approach.